packages/api/src/libs/osnap/utils.ts

```diff
@@ -0,0 +1,87 @@
+import assert from "assert";
+import axios from "axios";
+import { Network } from "@tenderly/sdk";
+import type { OsnapPluginData, Transaction } from "./interfaces";
+
+// a list of chains osnap is enabled on. must be updated manually as we add more
+export const ChainsEnabled: Network[] = [
+  Network.MAINNET,
+  Network.GOERLI,
+  Network.POLYGON,
+  Network.OPTIMISTIC,
+  Network.ARBITRUM_ONE,
+];
+
+export type TenderlyApiConfig = {
+  accountName: string;
+  projectName: string;
+  accessKey: string;
+};
+export type SimulationRequest = {
+  save: boolean;
+  save_if_fails: boolean;
+  simulation_type: "full" | "quick";
+  network_id: string;
+  from: string;
+  to: string;
+  input: string;
+  gas: number;
+  state_objects: {
+    [address: string]: {
+      storage: {
+        [key: string]: string;
+      };
+    };
+  };
+};
+
+export type SimulationResponse = {
+  simulation: {
+    id: string;
+    status: boolean;
+  };
+};
+
+export type SimulationBundleRequest = {
+  simulations: SimulationRequest[];
+};
+export class TenderlyApi {
+  constructor(private config: TenderlyApiConfig) {}
+  getTenderlyApiUrl(): string {
+    return `https://api.tenderly.co/api/v1/account/${this.config.accountName}/project/${this.config.projectName}/simulate`;
+  }
+  // this runs a single transaction simulation, tenderly can also support an array but this changes input/output
+  simulate = async (params: SimulationRequest): Promise<SimulationResponse> => {
```

Comment:
this is changed to be just a function now, but originally since it was in a class its away to autobind 'this' when pointing to the function. it was previously exposed here: https://github.com/UMAprotocol/protocol/pull/4684/files#diff-6011bababf1d1dff942475e1f051d5ec42c12b039ba6842d128c309e6cce093eR10